File: BaseLua/src/main/java/org/anime_game_servers/lua/engine/LuaValue.java
package org.anime_game_servers.lua.engine;

public interface LuaValue {
    boolean isNull();

    boolean isBoolean();

    boolean isInteger();

    boolean isLong();

    boolean isDouble();

    boolean isFloat();

    boolean isString();

    boolean isTable();

    boolean asBoolean();

    int asInteger();

    long asLong();

    double asDouble();

    float asFloat();

    String asString();

    <T> T asObject(Class<T> type);
}

File: base/src/commonMain/kotlin/org/anime_game_servers/lua/engine/LuaValue.kt
package org.anime_game_servers.lua.engine

interface LuaValue {
    fun isNull() : Boolean
    fun isBoolean() : Boolean
    fun isInteger() : Boolean
    fun isLong() : Boolean
    fun isDouble() : Boolean
    fun isFloat() : Boolean
    fun isString() : Boolean
    fun isTable() : Boolean
    fun asBoolean(): Boolean
    fun asInteger(): Int
    fun asLong(): Long
    fun asDouble(): Double
    fun asFloat(): Float
    fun asString(): String?
}
